addons/stock/models/stock_move_line.py

```diff
@@ -344,6 +346,11 @@ def write(self, vals):
                 move.product_uom_qty = move.quantity_done
         next_moves._do_unreserve()
         next_moves._action_assign()
+
+        if move_to_recompute_state:
```

Comment:
``` python
        # The user wrote on a a reserved move line and its `move_id` state needs to be recomputed.
        # As this write most probably happened in a On2many (where the `recompute` is delayed) and
        # the state of the move depends on the `product_qty` stored computed field on the move
        # line, we need to skip this optimisation and recompute directly here so that
        # `_compute_reserved_availability` works as expected.
```